Backend/sub_crates/language/src/tools/register.rs

```diff
@@ -20,34 +20,18 @@ impl Register for Dictionary {
             value_str = value_str.replace(&format!("{{{env_key}}}", env_key = key), &val);
         }
         let lang_index = language as usize;
-        match lang_table.get_mut(&key_str) {
-            Some(vec) => {
-                let vec_length = vec.len();
-                match vec.get_mut(lang_index) {
-                    Some(content) => {
-                        if content.is_empty() {
-                            *content = value_str;
-                        } else {
-                            panic!("{} is overwritten for the language {} with the content {}!", key, lang_index, value);
-                        }
-                    },
-                    // This means that the vector has not been extended to this language yet
-                    None => {
-                        for _ in vec_length..lang_index {
-                            vec.push(String::new());
-                        }
-                        vec.push(value_str);
-                    },
-                }
-            },
-            None => {
-                let mut container: Vec<String> = Vec::new();
-                for _ in 0..lang_index {
-                    container.push(String::new());
-                }
-                container.push(value_str);
-                lang_table.insert(key_str, container);
+        if !lang_table.contains_key(&key_str) {
+            lang_table.insert(String::from(&key_str), Vec::<Option<String>>::new());
+        }
+        let vec = lang_table.get_mut(&key_str).unwrap();
+        for _ in vec.len()..lang_index + 1 {
```

Comment:
Denke hier wäre es besser für alle Sprachen statt nur bis zur gewohlten nen "None" value anzulegen

Reply:
Dafür müsste ich wissen, wie viele Sprachen es gibt. Ich hab bisher keine (simple) Methode gefunden, die Anzahl der enum-EInträge zu finden.

Reply:
Wie in C/C++ fügst du hinten ein Eintrag hinzu

Reply:
Wie besprochen habe ichs mit strum gelöst, fand ich schöner. Wenn man schon eine saubere Sprache hat, muss man sich doch nicht die C-Hacks wieder reinholen :D
Guck mal bitte, ob ich den crate richtig eingefügt habe. Durch die Definition von Macros scheint das in der Wurzel nötig zu sein, zumindest hat mir das der Compiler so gesagt...